clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -43,6 +43,7 @@
 #include "llvm/Passes/PassBuilder.h"
 #include "llvm/Passes/PassPlugin.h"
 #include "llvm/Passes/StandardInstrumentations.h"
+#include "llvm/SYCLLowerIR/CleanupSYCLCompilerInternalMetadata.h"
```

Comment:
Do you mind giving some more information on the problem we are trying to solve?

Reply:
The problem is the following:
We split IR module multiple times in the code split and then trying to link the split modules using llvm-link. Since llvm-link tool just regularizes the metadata it creates a multiple copies of the same module metadata increasing module size considerably.